### DISCUSSION TITLE
JSX double quotes getting transformed to &quot;

### DISCUSSION THREAD

**Discussion by Jaikant** [kriasoft/react-starter-kit#2025] [Q&A]
In my JSX a double quote " is getting transformed to \&quot;

I am guessing it could be babel which is doing this? Any ideas? How do I retain the double quotes?

The double quote is part of a string.

Like this <script> {' some text " '} </script>

Notice the double quotes after 'some text' above

I tried to use a back tick ()` instead of the single quote for the string, but it did not help.

**Comment by Jaikant** [comment]
Just for context, I am introducing google structured data on my website. As in the link below:

[https://developers.google.com/search/docs/guides/intro-structured-data
](https://developers.google.com/search/docs/guides/intro-structured-data
)
I am passing the JSON-LD string from the index.js in my routes, and processing it in the Html.js 

The double quotes (") get transformed to \&quot; after the below in server.js

    const html = ReactDOM.renderToStaticMarkup(<Html {...data} />);

The tool which is here does not seem to support  \&quot; in place of "
[https://search.google.com/structured-data/testing-tool](https://search.google.com/structured-data/testing-tool
)

**Comment by Jaikant** [comment]
I resolved this by using dangerouslySetInnerHTML within the script tag.

**Comment by johan237** [comment]
Hey @Jaikant I am facing a similar problem in react and do not understand what you did please can you further explain with example perhaps

**Comment by koistya** [comment]
```jsx
<div dangerouslySetInnerHTML={{
  __html: `<p>Hello</p>`
}} />
```

```jsx
<script dangerouslySetInnerHTML={{
  __html: `console.log("Hello")`
}} />
```

https://reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml
